src/main/java/ch/njol/skript/effects/EffEnchant.java

```diff
@@ -36,72 +31,113 @@
+	"disenchant the player's tool from unbreaking and sharpness",
+	"",
+	"# For enchanted books",
+	"enchant player's tool with stored unbreaking 3",
+	"disenchant stored enchantments from player's tool"
+})
+@Since("2.0, INSERT VERSION (stored, specific disenchant)")
 public class EffEnchant extends Effect {
 	static {
 		Skript.registerEffect(EffEnchant.class,
-				"enchant %~itemtypes% with %enchantmenttypes%",
-				"disenchant %~itemtypes%");
+				"enchant %~itemtypes% with [:stored] %-enchantmenttypes%",
+				"disenchant %~itemtypes% [specific:(of|from) %-enchantmenttypes%]",
+				"disenchant (stored:stored (specific:%-enchantmenttypes%|enchant[ment]s) (of|from)) %~itemtypes%");
 	}
-	
+
 	@SuppressWarnings("null")
-	private Expression<ItemType> item;
+	private Expression<ItemType> items;
 	@Nullable
-	private Expression<EnchantmentType> enchs;
-	
-	@SuppressWarnings({"unchecked", "null"})
+	private Expression<EnchantmentType> enchantments;
+	private boolean isStored, isDisenchant, isSpecificDisenchant;
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		item = (Expression<ItemType>) exprs[0];
-		if (!ChangerUtils.acceptsChange(item, ChangeMode.SET, ItemStack.class)) {
-			Skript.error(item + " cannot be changed, thus it cannot be (dis)enchanted");
+	@SuppressWarnings({"unchecked", "null"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		items = (Expression<ItemType>) (matchedPattern == 2 ? exprs[1] : exprs[0]);
+
+		if (!ChangerUtils.acceptsChange(items, ChangeMode.SET, ItemStack.class)) {
+			Skript.error(items + " cannot be changed, thus it cannot be (dis)enchanted");
 			return false;
 		}
-		if (matchedPattern == 0)
-			enchs = (Expression<EnchantmentType>) exprs[1];
+		isStored = parseResult.tags.contains("stored");
+		isDisenchant = matchedPattern > 0;
+		isSpecificDisenchant = parseResult.tags.contains("specific");
+		enchantments = exprs.length == 2 ? (Expression<EnchantmentType>) exprs[(matchedPattern == 2 ? 0 : 1)] : null;
 		return true;
 	}
-	
+
 	@Override
-	protected void execute(final Event e) {
-		final ItemType i = item.getSingle(e);
-		if (i == null)
+	protected void execute(Event event) {
+		ItemType[] items = this.items.getArray(event);
+		if (items.length < 1)
 			return;
-		if (enchs != null) {
-			final EnchantmentType[] types = enchs.getArray(e);
-			if (types.length == 0)
+
+		EnchantmentType[] types = null;
+
+		if ((isSpecificDisenchant || !isDisenchant)) {
+			if (enchantments == null)
 				return;
-			
-			for (final EnchantmentType type : types) {
-				Enchantment ench = type.getType();
-				assert ench != null;
-				i.addEnchantments(new EnchantmentType(ench, type.getLevel()));
+
+			types = enchantments.getArray(event);
+			if (types.length < 1)
+				return;
+		}
+		
+		if (!isDisenchant) {
+			for (ItemType item : items) {
+				if (isStored) {
```

Comment:
This if statement doesn't need to exist when the only difference between both if statements is a method name